sycl/source/backend.cpp

```diff
@@ -246,10 +246,10 @@ kernel make_kernel(const context &TargetContext,
 
 kernel make_kernel(pi_native_handle NativeHandle, const context &TargetContext,
                    backend Backend) {
-  return make_kernel(TargetContext,
-                     get_kernel_bundle<bundle_state::executable>(
-                         TargetContext, std::vector<kernel_id>{}),
-                     NativeHandle, false, Backend);
+  return make_kernel(
+      TargetContext,
+      get_empty_interop_kernel_bundle<bundle_state::executable>(TargetContext),
```

Comment:
Please explain why empty kernel_bundle is needed and sufficient here?
Also add description for PR.

Reply:
> Please explain why empty kernel_bundle is needed and sufficient here?

Using an empty kernel bundle for this function was introduced in https://github.com/intel/llvm/commit/53ea8b9bc1e9b61e200ce467eeb1da868dcdaf60 to prevent building other kernels when working with interop. It is sufficient here because interop kernels don't currently have `kernel_id`s associated with them as there are no way of getting the `kernel_id` of a non-SYCL kernel in the current spec. Since the kernels are self-contained however that is fine.

Reply:
Ah, OK. I missed the fact that it was already empty "kernel_bundle" before this change. So, what's the improvement that this change is doing?

Reply:
In `enqueueImpKernel` this adds another condition to taking the path using the associated `kernel_bundle`, namely that interop bundles can't be used in the first branch. This is because the kernels in interop kernel bundles (if any) do not have `kernel_id` and can therefore not be looked up, but since they are self-contained they can simply be launched directly.

Reply:
Thanks for explaining! This clarification should go as a comment around the code that git new "if" condition.

Reply:
Comment added.